designcompose/src/testDebug/kotlin/com/android/designcompose/DesignSwitcherBasicTests.kt

```diff
@@ -0,0 +1,90 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.designcompose
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.test.hasClickAction
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.android.designcompose.test.internal.captureRootRoboImage
+import com.android.designcompose.test.internal.designComposeRoborazziRule
+import com.android.designcompose.test.onDCDoc
+import com.github.takahirom.roborazzi.RobolectricDeviceQualifiers
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.TestName
+import org.junit.runner.RunWith
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.GraphicsMode
+
+@Composable
+fun DesignSwitcherTest(testName: TestName) {
+    val idState = remember { mutableStateOf(testName.methodName) }
+    DesignSwitcher(doc = null, currentDocId = idState.value, branchHash = null, setDocId = {})
+}
+
+@RunWith(AndroidJUnit4::class)
+@GraphicsMode(GraphicsMode.Mode.NATIVE)
+@Config(qualifiers = RobolectricDeviceQualifiers.SmallPhone, sdk = [34])
+class DesignSwitcherBasicTests {
+    @get:Rule val composeTestRule = createComposeRule()
+    @get:Rule val testName = TestName()
+    @get:Rule val roborazziRule = designComposeRoborazziRule(javaClass.simpleName)
+
+    @Before
+    fun resetState() {
+        // Must reset the interaction state so that the DesignSwitcher starts off closed.
+        InteractionStateManager.states.clear()
+        composeTestRule.setContent { DesignSwitcherTest(testName = testName) }
+    }
+
+    @Test
+    fun canExpandDesignSwitcher() {
+        with(composeTestRule) {
+            captureRootRoboImage("CollapsedSwitcher")
+            onDCDoc(DesignSwitcherDoc).performClick()
+            onNodeWithText("Change").assertExists()
+            captureRootRoboImage("ExpandedSwitcher")
+        }
+    }
+
+    @Test
+    fun showChangeScreen() {
+        with(composeTestRule) {
+            onDCDoc(DesignSwitcherDoc).performClick()
+            onNodeWithText("Change", useUnmergedTree = true).performClick()
+            onNodeWithText("Load", useUnmergedTree = true).assertExists()
+            captureRootRoboImage("ChangeFileScreen")
+        }
+    }
+
+    @Test
+    fun canCheckACheckBox() {
+        with(composeTestRule) {
+            onDCDoc(DesignSwitcherDoc).performClick()
+            onNodeWithText("Options", useUnmergedTree = true).performClick()
+            captureRootRoboImage("OptionsScreenBeforeCheckingBox")
+            onAllNodes(hasClickAction())[2].performClick() // Currently the only way to find it
```

Comment:
This finds the third node that has a click action and clicks it?

Reply:
Yep. The followup (#797) will be replacing this line with one that looks for the specific node name.